contributed_definitions/nyaml/NXopt.yaml

```diff
@@ -56,348 +43,675 @@ NXopt(NXobject):
+          
+          The window is considered to be a part of the sample stage but also
+          beam path. Hence, its position within the beam path should be
+          defined by the 'depends_on' field.
+        window_effects_corrected(NX_BOOLEAN):
+          doc: |
+            Was a window correction performed? If 'True' describe the window
+            correction procedure in 'window_correction/procedure'.
+        window_correction(NXprocess):
+          exists: optional
+          doc: |
+            Was a window correction performed? If 'True' describe the
+            window correction procedure in ''
+          procedure:
             doc: |
-              The material of the window.
-            enumeration: [quartz, diamond, calcium fluoride, zinc selenide, thallium bromoiodide, alkali halide compound, Mylar, other]
-          other_material(NX_CHAR):
+              Describe when (before or after the main measurement + time
+              stamp in 'date') and how the window effects have been
+              corrected, i.e. either mathematically or by performing a
+              reference measurement. In the latter case, provide the link to
+              to the reference data in 'reference_data_link'.
+          reference_data_link:
             exists: optional
             doc: |
-              If you specified 'other' as material, decsribe here what it is.
-          thickness(NX_FLOAT):
-            unit: NX_LENGTH
-            doc: |
-              Thickness of the window.
-          orientation_angle(NX_FLOAT):
-            unit: NX_ANGLE
-            doc: |
-              Angle of the window normal (outer) vs. the substrate normal
-              (similar to the angle of incidence).
+              Link to the NeXus file which describes the reference data if a
+              reference measurement for window correction was performed.
+              Ideally, the reference measurement was performed on a reference
+              sample and on the same sample, and using the same conditions as
+              for the actual measurement with and without windows. It should
+              have been conducted as close in time to the actual measurement
+              as possible.
+        material(NX_CHAR):
+          doc: |
+            The material of the window.
+          enumeration: [quartz, diamond, calcium fluoride, zinc selenide, thallium bromoiodide, alkali halide compound, Mylar, other]
+        material_other(NX_CHAR):
+          exists: optional
+          doc: |
+            If you specified 'other' as material, decsribe here what it is.
+        thickness(NX_FLOAT):
+          unit: NX_LENGTH
+          doc: |
+            Thickness of the window.
+        orientation_angle(NX_FLOAT):
+          unit: NX_ANGLE
+          doc: |
+            Angle of the window normal (outer) vs. the substrate normal
+            (similar to the angle of incidence).
+      sample_medium_refractive_indices(NX_FLOAT):
```

Comment:
is this part of the sample?

Reply:
was there before, and can be discussed. I felt like it related better to the setup, as it may be on additional beam device, which has specific transmission properties.